src/js/worklets/communities.js

```diff
@@ -79,55 +104,81 @@ export function useSheetStyles({ scrollAmount, expandHeaderThreshold, sheetDispl
   });
 }
 
-export function useNameStyles({ scrollAmount, expandHeaderThreshold, textMovementThreshold }) {
+export function useNameStyles({ initialState, scrollAmount, collapseThreshold, textMovementThreshold }) {
   return useAnimatedStyle(() => {
-    const animationProgress = interpolate(
-      scrollAmount.value,
-      [textMovementThreshold, expandHeaderThreshold],
-      [0, 40],
-      'clamp',
-    );
+    let horizontalPosition;
+    if (initialState.value === 'collapsed') {
+      horizontalPosition = 40;
+    } else if (initialState.value === 'expanded') {
+      horizontalPosition = 0;
+    } else {
+      horizontalPosition = interpolate(
+        scrollAmount.value,
+        [textMovementThreshold.value, collapseThreshold.value],
+        [0, 40],
+        'clamp',
+      );
+    }
+
+    let verticalPosition;
+    if (initialState.value === 'collapsed') {
+      verticalPosition = -44.5;
+    } else if (initialState.value === 'expanded') {
+      verticalPosition = 0;
+    } else {
+      verticalPosition = interpolate(
+        scrollAmount.value,
+        [textMovementThreshold.value, collapseThreshold.value],
+        [0, -44.5],
+        'clamp',
+      );
+    }
+
     return {
-      marginRight: animationProgress,
-      transform: [
-        { translateX: animationProgress },
-        {
-          translateY: interpolate(
-            scrollAmount.value,
-            [textMovementThreshold, expandHeaderThreshold],
-            [0, -44.5],
-            'clamp',
-          ),
-        },
-      ],
+      marginRight: horizontalPosition,
+      transform: [{ translateX: horizontalPosition }, { translateY: verticalPosition }],
     };
   });
 }
 
-export function useInfoStyles({ scrollAmount, infoOpacityThreshold }) {
+export function useInfoStyles({ initialState, scrollAmount, collapseThreshold, infoOpacityThresholdFactor }) {
   return useAnimatedStyle(() => {
+    let opacity;
+    if (initialState.value === 'collapsed') {
+      opacity = 0;
+    } else if (initialState.value === 'expanded') {
+      opacity = 1;
```

Comment:
Not sure who is choosing the names for `collapsed` or `expanded` (whether it's us or a library), but maybe we can introduce some constants for the strings?